src/lib/components/Swap/Settings/MaxSlippageSelect.tsx

```diff
@@ -98,61 +98,69 @@ const Warning = memo(function Warning({ state, showTooltip }: { state: WarningSt
-    const numerator = Math.floor(+custom * 100)
-    if (numerator) {
-      const percent = new Percent(numerator, 10_000)
-      if (!percent.lessThan(MAX_VALID_SLIPPAGE)) {
-        setWarning(WarningState.INVALID_SLIPPAGE)
-        setMaxSlippage('auto')
-      } else if (!percent.lessThan(MIN_HIGH_SLIPPAGE)) {
-        setWarning(WarningState.HIGH_SLIPPAGE)
-        setMaxSlippage(percent)
+  const processInput = useCallback(
+    (input: number | undefined) => {
+      const numerator = input && Math.floor(input * 100)
+      if (numerator) {
+        const percent = new Percent(numerator, 10_000)
+        if (!percent.lessThan(MAX_VALID_SLIPPAGE)) {
+          setWarning(WarningState.INVALID_SLIPPAGE)
+          setAutoSlippage(true)
+          setMaxSlippage(input)
+        } else if (!percent.lessThan(MIN_HIGH_SLIPPAGE)) {
+          setWarning(WarningState.HIGH_SLIPPAGE)
+          setAutoSlippage(false)
+          setMaxSlippage(input)
+        } else {
+          setWarning(WarningState.NONE)
+          setAutoSlippage(false)
+          setMaxSlippage(input)
+        }
       } else {
-        setWarning(WarningState.NONE)
-        setMaxSlippage(percent)
+        setAutoSlippage(true)
+        setMaxSlippage(undefined)
       }
-    } else {
-      setMaxSlippage('auto')
-    }
-  }, [custom, setMaxSlippage])
-  useEffect(processInput, [processInput])
+    },
+    [setAutoSlippage, setMaxSlippage]
+  )
   const onInputSelect = useCallback(() => {
     focus()
-    processInput()
-  }, [focus, processInput])
+    processInput(maxSlippage)
+  }, [focus, maxSlippage, processInput])
+
+  useEffect(() => processInput(maxSlippage), [maxSlippage, processInput]) // processes any warnings on mount
+  useEffect(() => setShowTooltip(true), [warning, setShowTooltip]) // enables the tooltip if a warning is set
 
   return (
     <Column gap={0.75}>
       <Label name={<Trans>Max slippage</Trans>} tooltip={tooltip} />
       <Row gap={0.5} grow="last">
-        <Option wrapper={Button} selected={maxSlippage === 'auto'} onSelect={() => setMaxSlippage('auto')}>
+        <Option wrapper={Button} selected={autoSlippage} onSelect={() => setAutoSlippage(true)}>
           <ThemedText.ButtonMedium>
             <Trans>Auto</Trans>
           </ThemedText.ButtonMedium>
         </Option>
         <Option
           wrapper={Custom}
-          selected={maxSlippage !== 'auto'}
+          selected={!autoSlippage}
           onSelect={onInputSelect}
           icon={<Warning state={warning} showTooltip={showTooltip} />}
           {...tooltipProps}
         >
           <Row color={warning === WarningState.INVALID_SLIPPAGE ? 'error' : undefined}>
             <DecimalInput
-              size={Math.max(custom.length, 3)}
-              value={custom}
-              onChange={setCustom}
+              size={Math.max(value.length, 3)}
+              value={value}
+              onChange={(input) => processInput(+input)}
```

Comment:
nit - kinda feels like the typing logic should be happening in the processInput function

Reply:
`DecimalInput` takes a `string`, but I still want to store `maxSlippage` as a number so it is better typed. Making `processInput` take a `number` lets it be reused for `onInputSelect`/mounting logic as well.